plugins/fetchinvoice.c

```diff
@@ -620,6 +611,31 @@ static struct command_result *send_message(struct command *cmd,
 	return send_outreq(cmd->plugin, req);
 }
 
+/* We've received neither a reply nor a payment; return failure. */
+static void timeout_sent_inv(struct sent *sent)
+{
+	struct json_out *details = json_out_new(sent);
+
+	json_out_addstr(details, "invstring", invoice_encode(tmpctx, sent->inv));
+	/* This will free sent! */
+	discard_result(command_done_err(sent->cmd, OFFER_TIMEOUT,
+					"Timeout waiting for response"
+					" (but use waitinvoice if invoice_timeout"
+					" was greater)",
+					details));
```

Comment:
This is pretty unergonomic, better would be to `waitinvoice` internally if we know that there should be a payment coming.

Reply:
We can use the CLI notifications support to deliver these intermediate results.

Reply:
The idea was that you can explicitly set invoice timeout > the waiting timeout.  But in practice it's a little silly, since we don't do retransmission or anything.  So I'll move it back to a single timeout value.